src/Components/Components/src/BindConverter.cs

```diff
@@ -0,0 +1,1047 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Performs conversions during binding.
+    /// </summary>
+    //
+    // Perf: our conversion routines present a regular API surface that allows us to specialize on types to avoid boxing.
+    // for instance, many of these types could be cast to IFormattable to do the appropriate formatting, but that's going
+    // to allocate. 
+    public static class BindConverter
```

Comment:
Let's agree on the overall API shape and then I'll add docs. I'm not totally sure that we need unit tests for all of these because they are typically trivial and are covered by our bind E2E tests.

Also, most of the **parsing** code is cut-paste from the binding code.

Reply:
I agree on the API shape.

I'd think that unit tests would be *nice* because there are potentially a lot of different cases for handling unexpected values (e.g., non-defined enum entries, floats out of range, etc.) and as these become discovered, it's much cheaper to add unit tests than E2E tests.

Not saying you should immediately write a complete suite of unit tests covering every possible branch point in this whole file, since like you say most are trivial and the majority of behaviors are covered in E2E tests. But if there was at least a place for unit tests to go, and whatever small number of cases cover the more complex format/parsing methods here that might not be under E2E test already, that would make it cheap for other developers to add more unit test cases later if we want.